packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1688,7 +1688,10 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     Clipboard.setData(ClipboardData(text: selection.textInside(text)));
     _replaceText(ReplaceTextIntent(textEditingValue, '', selection, cause));
     if (cause == SelectionChangedCause.toolbar) {
-      bringIntoView(textEditingValue.selection.extent);
+      // Schedule a call to bringIntoView() after renderEditable updates.
+      SchedulerBinding.instance.addPostFrameCallback((_) {
+        bringIntoView(textEditingValue.selection.extent);
```

Comment:
This problem where RenderEditable is out of date is a recurring problem.  I think using an addPostFrameCallback is the best option for now.

Reply:
Indeed, it is a bit prone to more bugs. For `bringIntoView()` itself, I can see it's called from:
- text_selection.dart: `handleSelectAll()`, `_handleSelectionHandleChanged()`
- [material|cupertino] text_field.dart: `_handleSelectionChanged()`
-  selectable_text: `_handleSelectionChanged()`
- editable_text: `copySelection()`, `cutSelection()`, `pasteText()`, `selectAll()`, `_scrollToDocumentBoundary()`, `_updateSelection()`, `_expandSelection()`

The only two calls that may modify text contents/layout are cut and paste, handled by this PR.

BTW, another question (unrelated to this PR) is whether only calling `bringIntoView()` on SelectionChangedCause.toolbar is desired behaviour. This means that if one cuts/pastes in an area outside the viewport, we don't scroll there so the user doesn't see what's been done to the text.

Reply:
Glad that this PR will handle the problem here at least.

About scrolling on paste, I think you're right that it's incorrect.  I've created an issue: https://github.com/flutter/flutter/issues/98675.